src/constructors.ts

```diff
@@ -0,0 +1,68 @@
+import { Result, ResultValues, AggregateResponse, Results } from "./results";
+
+/**
+ * valueFromProto unpacks a proto value.
+ */
+export const valueFromProto = (proto: any): any => {
+  if (proto.single !== undefined) {
+    return proto.single;
+  }
+  if (proto.repeated.values !== undefined) {
+    return proto.repeated.values;
+  }
+  return null;
+};
+
+/**
+ * newResult constructs a Result from a proto Result.
+ */
+export const newResult = (resultJSON: any): Result => {
+  const values: ResultValues = {};
+  Object.keys(resultJSON.values).forEach(k => {
+    values[k] = valueFromProto(resultJSON.values[k]);
+  });
+  return {
+    values,
+    tokens: {},
+    score: parseFloat(resultJSON.score),
+    indexScore: parseFloat(resultJSON.indexScore)
+  };
+};
+
+/**
+ * newAggregates constructs an AggregateResponse object from proto
+ */
+export const newAggregates = (aggregateJSON: any = {}): AggregateResponse =>
+  Object.keys(aggregateJSON).reduce((prev: AggregateResponse, cur) => {
+    const [aggregateType, field] = cur.split(".");
+    if (aggregateType === "count") {
+      prev[cur] = aggregateJSON[cur].count.counts;
+    }
+    // todo(tbillington): implement bucket
```

Comment:
There is a [Bucket impl](https://github.com/sajari/sdk-node/blob/4f37e2169608c1e6f3bb6a4e84c61925b67f868e/src/pipeline/search.ts#L165) in the node SDK.

Reply:
I'm half a mind to leave this unimplemented. It's not currently implemented in the api so I don't have a good way to test it. It will be easy add support once it's available in the api.

Reply:
@dhowden said he could help me test so will investigate 🤓